livekit_ingress.proto

```diff
@@ -0,0 +1,90 @@
+syntax = "proto3";
+
+package livekit;
+
+import "google/protobuf/empty.proto";
+import "livekit_models.proto";
+
+option go_package = "github.com/livekit/protocol/livekit";
+
+service Ingress {
+    rpc CreateIngress(CreateIngressRequest) returns (IngressInfo);
+    rpc ListIngress(ListIngressRequest) returns (ListIngressResponse);
+    rpc DeleteIngress(DeleteIngressRequest) returns (google.protobuf.Empty);
+}
+
+enum IngressInput {
+    RTMP_INPUT = 0;
+//    FILE_INPUT = 1;
+//    SRT_INPUT = 2;
+//    URL_INPUT = 3;
+}
+
+message IngressInfo {
+    enum State {
+        ENDPOINT_WAITING = 0;
+        ENDPOINT_ACTIVE = 1;
+        ENDPOINT_ENDED = 2;
+    }
+    string id = 1;
+    IngressInput input_type = 2;
+    State state = 3;
+    string room = 4;
+    string participant_identity = 5;
+    string participant_name = 6;
+    bool transcode = 7;
+    // for RTMP input, it'll be a rtmp:// URL
+    // for FILE input, it'll be a http:// URL
+    // for SRT input, it'll be a srt:// URL
+    string url = 8;
+    repeated TrackInfo tracks = 9;
+}
+
+message CreateIngressRequest {
+    IngressInput input_type = 1;
+    // room to publish to
+    string room_name = 2;
+    // publish as participant
+    string participant_identity = 3;
+    // name of publishing participant (used for display only)
+    string participant_name = 4;
+    IngressAudioOptions audio = 5;
+    IngressVideoOptions video = 6;
+    // true to disable transcoding and publish input codecs
+    // when this is enabled, and input codecs aren't compatible with WebRTC, we'll
+    // throw an error.
+    bool disable_transcode = 7;
+}
+
+message IngressAudioOptions {
+    string name = 1;
+    TrackSource source = 2;
+    // desired mime_type to publish to room
+    string mime_type = 3;
+    uint32 bitrate = 4;
+    bool dtx = 5;
+    uint32 channels = 6;
+}
+
+message IngressVideoOptions {
+    string name = 1;
+    TrackSource source = 2;
+    // desired mime_type to publish to room
+    string mime_type = 3;
+    // simulcast layers to publish, when empty, it'll pick default simulcast
+    // layers at 1/2 and 1/4 of the dimensions
+    repeated VideoLayer layers = 4;
```

Comment:
Guessing the idea is to encode simulcast at the bridge? Also, should we encode simulcast by default? Guess it does make sense especially to have layers if we are doing SVC.

Reply:
yeah, I think we should enable simulcast by default since it'll be needed to ensure uninterrupted streams for lower bandwidth users. 

The way to override is just to publish a single layer.